src/http-client.ts

```diff
@@ -43,9 +43,13 @@ export class HttpClient {
   requestApi (path: string, opts: RequestApiOptions): Promise<any> {
     let response = request[opts.method](this._buildUri(opts.host, path, opts.version))
       .use(reqlogger.bind(this, opts.logger, opts.data))
-      .set('Authorization', `Bearer ${opts.authToken}`)
       .type(opts.contentType || 'json');
 
+    // default to include auth header
+    if (!opts.noAuthHeader) {
+      response.set('Authorization', `Bearer ${opts.authToken}`);
+    }
```

Comment:
I'm going to be using this http-client in the sdk and the sdk needs to be able to make requests without an authorization header.